cmd/solana_exporter/slots.go

```diff
@@ -89,44 +89,78 @@ func (c *solanaCollector) WatchSlots() {
 	if err != nil {
 		klog.Error(err)
 	}
-	ticker := time.NewTicker(slotPacerSchedule)
+	ticker := time.NewTicker(c.slotPace)
 
 	for {
-		<-ticker.C
+		select {
+		case <-ctx.Done():
+			klog.Infof("Stopping WatchSlots() at slot %v", watermark)
+			return
 
-		// Get current slot height and epoch info
-		ctx, cancel := context.WithTimeout(context.Background(), httpTimeout)
-		info, err := c.rpcClient.GetEpochInfo(ctx, rpc.CommitmentMax)
-		if err != nil {
-			klog.Infof("failed to fetch epoch info, retrying: %v", err)
+		default:
+			<-ticker.C
+
+			// Get current slot height and epoch info
+			ctx_, cancel := context.WithTimeout(context.Background(), httpTimeout)
+			info, err := c.rpcClient.GetEpochInfo(ctx_, rpc.CommitmentMax)
+			if err != nil {
+				klog.Infof("failed to fetch epoch info, retrying: %v", err)
```

Comment:
I realize you're just moving the code around, but consider making this a warning as an error isn't really an info.

Reply:
will do